src/AbstractRoboFile.php

```diff
@@ -471,70 +476,146 @@ protected function syncTask(
+            $opts['files'] = false;
+
+            $tmpKeyFile = '~/.ssh/' . uniqid('robo_', true) . '.id_rsa';
+
+            // Generate a temporary key.
+            $collection->addTask(
+                $this->taskExec('ssh-keygen -q -t rsa -b 4096 -N "" -f ' . $tmpKeyFile . ' -C "robo:' . md5($tmpKeyFile) . '"')
+            );
+
+            $collection->completion(
+                $this->taskExecStack()
+                    ->exec('rm -f ' . $tmpKeyFile . ' ' . $tmpKeyFile . '.pub')
+            );
+
+            // Install it on the destination host.
+            $collection->addTask(
+                $this->taskExec('cat ' . $tmpKeyFile . '.pub | ssh ' . $destinationUser . '@' . $destinationHost . ' -o StrictHostKeyChecking=no -i ' . $destinationKeyFile . ' "mkdir -p ~/.ssh && cat >> ~/.ssh/authorized_keys"')
+            );
+
+            $collection->completion(
+                $this->taskSsh($destinationHost, $destinationAuth)
+                    ->exec('sed -i "/robo:' . md5($tmpKeyFile) . '/d" ~/.ssh/authorized_keys')
+            );
+
+            $collection->addTask(
+                $this->taskRsync()
+                    ->rawArg('--rsh "ssh -o StrictHostKeyChecking=no -i `readlink -e ' . $sourceKeyFile . '`"')
+                    ->fromPath($tmpKeyFile)
+                    ->toHost($sourceHost)
+                    ->toUser($sourceUser)
+                    ->toPath('~/.ssh')
+                    ->archive()
+                    ->compress()
+                    ->checksum()
+                    ->wholeFile()
+            );
+
+            $collection->completion(
+                $this->taskSsh($sourceHost, $sourceAuth)
+                    ->exec('rm -f ' . $tmpKeyFile)
+            );
+
+            $dirs = ($this->fileBackupSubDirs ? $this->fileBackupSubDirs : ['']);
+
+            foreach ($dirs as $dir) {
+                $dir .= ($dir !== '' ? '/' : '');
+
+                $rsync = $this->taskRsync()
+                    ->rawArg('--rsh "ssh -o StrictHostKeyChecking=no -i `readlink -e ' . $tmpKeyFile . '`"')
+                    ->fromPath($sourceRemote['filesdir'] . '/' . $dir)
+                    ->toHost($destinationHost)
+                    ->toUser($destinationUser)
+                    ->toPath($destinationRemote['filesdir'] . '/' . $dir)
+                    ->archive()
+                    ->delete()
+                    ->rawArg('--copy-links --keep-dirlinks')
+                    ->compress()
+                    ->checksum()
+                    ->wholeFile();
+
+                foreach ($this->excludeFromBackup as $exclude) {
+                    $rsync->exclude($exclude);
+                }
+
+                $collection->addTask(
+                    $this->taskSsh($sourceHost, $sourceAuth)
+                        ->timeout(300)
+                        ->exec($rsync)
+                );
+            }
+        }
+
+        if ($opts['data'] || $opts['files']) {
```

Comment:
DOn't we have to split this up so if rsync is enabled, files aren't backupped/ synced like this?